### PR TITLE
Scala3doc: Add support for annotations in type parameters, add tests for external location providing, minor fixes

### PR DESCRIPTION
**Pull request by pikinier20**
closes [#250](https://github.com/lampepfl/scala3doc/issues/250)
closes [#111](https://github.com/lampepfl/scala3doc/issues/111)

### CODE REVIEW

File: scala3doc-testcases/src/tests/specializedSignature.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package tests
+
+package specializedSignature
+
+import scala.{specialized}
+
+trait AdditiveMonoid[@specialized(Int, Long, Float, Double) A]
+{
+  def a: A
```
Comment (romanowski) [inline scala3doc-testcases/src/tests/specializedSignature.scala:9]: can we also add a case where `@specialized` is added to a function?

File: scala3doc/test/dotty/dokka/ExternalLocationProviderTest.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  def testResolvedLinks(provider: ScalaExternalLocationProvider, testcases: List[(DRI, String)]) = testcases.foreach {
+    case (dri, expect) => assertEquals(provider.resolve(dri), expect)
+  }
+
+  @Test
+  def javadocExternalLocationProviderTest(): Unit = {
+    val provider = createExternalLocationProvider("https://docs.oracle.com/javase/8/docs/api/", ".html", DocumentationKind.Javadoc)
+    val testcases = List(
+      (DRI("java.util.Map$$Entry"), "https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html"),
```
Comment (romanowski) [inline scala3doc/test/dotty/dokka/ExternalLocationProviderTest.scala:47]: shouldn't we add a tests for method base `DRI`?

**Review by romanowski — CHANGES_REQUESTED** [review]
Could we also create a basic integration test for external links? One using Jsoup and parsing generated htmls

File: scala3doc/test/dotty/dokka/ExternalLocationProviderTest.scala (hunk 2 above)
Comment (pikinier20) [reply scala3doc/test/dotty/dokka/ExternalLocationProviderTest.scala:47]: I'm gonna make task for that because linking to methods (anchors) are still TODO

Comment (pikinier20) [reply scala3doc/test/dotty/dokka/ExternalLocationProviderTest.scala:47]: https://github.com/lampepfl/scala3doc/issues/263

File: scala3doc-testcases/src/tests/specializedSignature.scala (hunk 1 above)
Comment (pikinier20) [reply scala3doc-testcases/src/tests/specializedSignature.scala:9]: Done

**Comment by pikinier20** [comment]
Added integration tests. @romanowski

**Review by romanowski — APPROVED** [review]
I really like how this PR looks now.

**Pull request merged**